romana/kubernetes/listener.go

```diff
@@ -0,0 +1,364 @@
+//			log.Println("Exiting goroutine")
+//			return
+//		}
+//		if len(chunk) != 0 {
+//			time.Sleep(1 * time.Millisecond)
+//		} else {
+//			ch <- chunk
+//		}
+//	}
+//}
+
+// processChunk processes a chunk received from Kubernetes. A chunk
+// is a new update.
+//func processChunk(chunk []byte) error {
+//	m := make(map[string]interface{})
+//	err := json.Unmarshal(chunk, &m)
+//	if err != nil {
+//		return err
+//	}
+//	method := m["method"]
+//	if method == nil {
+//		return errors.New("Expected 'method' field")
+//	}
+//	methodStr = strings.ToLower(method.(string))
+//	networkPolicyIfc := m["policy_definition"]
+//	if networkPolicyIfc == nil {
+//		return errors.New("Expected 'policy_definition' field")
+//	}
+//	kubeNetworkPolicy := networkPolicyIfc.(map[string]interface{})
+//	romanaNetworkPolicy, err := translateNetworkPolicy(kubeNetworkPolicy)
+//	if err != nil {
+//		return err
+//	}
+//
+//	if methodStr == "added" {
+//		applyNetworkPolicy(networkPolicyActionAdd, romanaNetworkPolicy)
+//	} else if methodStr == "deleted" {
+//		applyNetworkPolicy(networkPolicyActioDelete, romanaNetworkPolicy)
+//	} else if methodStr == "modified" {
+//		applyNetworkPolicy(networkPolicyActioModify, romanaNetworkPolicy)
+//	} else {
+//		return common.NewError("Unexpected method '%s'", methodStr)
+//	}
+//}
+
+// Run implements the main loop, reconnecting as needed.
+//func RunListener0(rootURL string, kubeURL string) {
+//	l := kubeListener{kubeURL: kubeURL, restClient: common.NewRestClient(common.GetDefaultRestClientConfig(rootURL))}
+//	l.Run()
+//}
+//
+//func (l kubeListener) Run0() {
+//	for {
+//		ch := make(chan []byte)
+//		go listen(ch, l.kubernetesURL)
+//		for {
+//			chunk, ok := <-ch
+//			log.Println(ok)
+//			if chunk != "" {
+//				log.Printf("Read chunk %v\n%s\n------------", ok, string(chunk))
+//				err = processChunk(chunk)
+//				if err != nil {
+//					// TODO is there any other way to handle this?
+//					log.Printf("Error processing chunk %s: %v", string(chunk), err)
+//				}
+//			}
+//			if !ok {
+//				break
+//			}
+//		}
+//	}
+//}
+
+/////////////////////////////////////////////////////////////
+
+type networkPolicyAction int
+
+const (
+	networkPolicyActionDelete networkPolicyAction = iota
+	networkPolicyActionAdd
+	networkPolicyActionModify
+)
+
+type kubeListener struct {
+	config           common.ServiceConfig
+	restClient       common.RestClient
+	kubeUrl          string
+	urlPrefix        string
+	segmentLabelName string
+}
+
+// Routes returns various routes used in the service.
+func (l *kubeListener) Routes() common.Routes {
+	routes := common.Routes{}
+	return routes
+}
+
+// Name implements method of Service interface.
+func (l *kubeListener) Name() string {
+	return "kubernetes-listener"
+}
+
+// SetConfig implements SetConfig function of the Service interface.
+func (l *kubeListener) SetConfig(config common.ServiceConfig) error {
+	m := config.ServiceSpecific
+	l.kubeUrl = m["kubernetes_url"].(string)
```

Comment:
Here and in line 191 (and possibly other places) we should use pre-defined constants, rather than string literals.